src/test/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilterUnitTest.java

```diff
@@ -0,0 +1,427 @@
+                partitionBounds, 100, 10, featuresTestData.coverage);
+    }
+
+    private static IntHistogram.CDF cumulativeCountsToCDF(final long[] cumulativeCounts) {
+        final float[] cdfFractions = new float[cumulativeCounts.length];
+        final long totalObservations = cumulativeCounts[cumulativeCounts.length - 1];
+        for(int index = 0; index < cdfFractions.length; ++index) {
+            cdfFractions[index] = cumulativeCounts[index] / (float)totalObservations;
+        }
+        return new IntHistogram.CDF(cdfFractions, totalObservations);
+    }
+
+    @Test(groups = "sv")
+    protected void testLocalXGBoostClassifierAccuracy() {
+        // check accuracy: predictions are same as classifierAccuracyData up to tolerance
+        AssertJUnit.assertArrayEquals("Probabilities predicted by classifier do not match saved correct answers",
+                predictYProbaSerial, classifierAccuracyData.yProba, probabilityTol);
+    }
+
+    @Test(groups = "sv")
+    protected void testLocalXGBoostClassifierSpark() {
+        final Predictor localPredictor = XGBoostEvidenceFilter.loadPredictor(localClassifierModelFile);
+        // get spark ctx
+        final JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        // parallelize classifierAccuracyData to RDD
+        JavaRDD<FVec> testFeaturesRdd = ctx.parallelize(Arrays.asList(classifierAccuracyData.features));
+        // predict in parallel
+        JavaDoubleRDD predictYProbaRdd
+                = testFeaturesRdd.mapToDouble(f -> localPredictor.predictSingle(f, false, 0));
+        // pull back to local array
+        final double[] predictYProbaSpark = predictYProbaRdd.collect()
+                .stream().mapToDouble(Double::doubleValue).toArray();
+        // check probabilities from spark are identical to serial
+        AssertJUnit.assertArrayEquals("Probabilities predicted in spark context differ from serial",
+                predictYProbaSpark, predictYProbaSerial, 0.0);
+    }
+
+    @Test(groups = "sv")
+    protected void testResourceXGBoostClassifier() {
+        // load classifier from resource
+        final Predictor resourcePredictor = XGBoostEvidenceFilter.loadPredictor(resourceClassifierModelFile);
+        final double[] resourceYProba = predictProba(resourcePredictor, classifierAccuracyData.features);
+        // check that predictions from resource are identical to local
+        AssertJUnit.assertArrayEquals("Predictions via loading predictor from resource is not identical to local file",
+                resourceYProba, predictYProbaSerial, 0.0);
+    }
+
+    @Test(groups = "sv")
+    protected void testFeatureConstruction() {
+        final XGBoostEvidenceFilter evidenceFilter = new XGBoostEvidenceFilter(
+                evidenceList.iterator(), readMetadata, params, emptyCrossingChecker
+        );
+        for(int ind = 0; ind < featuresTestData.stringReps.length; ++ind) {
```

Comment:
This seems like a good place for a `DataProvider`.

Reply:
I'm not sure how to use a DataProvider here. The data is already aggregated and available, I'm just looping through to check if it's consistent.